pkg/cmd/cmd.go

```diff
@@ -61,6 +62,56 @@ func Uninstall(ctx context.Context, k8sAPI *k8s.KubernetesAPI, selector string)
 	return nil
 }
 
+// Prune takes an install manifest and prints all resources on the cluster which
+// match the given label selector but are not in the given manifest. Users are
+// expected to pipe these resources to `kubectl delete` to clean up resources
+// left on the cluster which are no longer part of the install manifest.
+func Prune(ctx context.Context, k8sAPI *k8s.KubernetesAPI, expectedManifests string, selector string) error {
+	expectedResources := []resource.Kubernetes{}
+	yamls := strings.Split(expectedManifests, "\n---\n")
+	for _, manifest := range yamls {
+		resource := resource.Kubernetes{}
+		err := yaml.Unmarshal([]byte(manifest), &resource)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error parsing manifest: %s", manifest)
+			os.Exit(1)
+		}
+		expectedResources = append(expectedResources, resource)
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: selector,
+	}
+	resources, err := resource.FetchPrunableResources(ctx, k8sAPI, metav1.NamespaceAll, listOptions)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error fetching resources: %s\n", err)
+		os.Exit(1)
```

Comment:
The pattern that we follow for CLI code is that anything which is a user-error should return an error, which will cause cobra to print the error as well as the usage help text.  For anything which is not a user error (e.g. 5XX from the API) we use `os.Exit(1)` directly to bypass Cobra so that it doesn't print usage text.